Scripts/RemediationScripts/Remediate-EnableTransparentDataEncryption.ps1

```diff
@@ -0,0 +1,653 @@
+                
+                if ($tdeStatus -eq 'Enabled')
+                {
+                    $sqlServersRemediated += $databaseDetail
+                }
+                else
+                {
+                    $sqlServersSkipped += $databaseDetail
+                }
+            }
+            catch
+            {
+                $sqlServersSkipped += $databaseDetail
+            }
+            
+        }
+
+        Write-Host $([Constants]::SingleDashLine)
+
+        if (($sqlServersRemediated | Measure-Object).Count -eq $totalSqlDatabaseWithTDEDisabled)
+        {
+            Write-Host "Transparent Data Encryption (TDE) successfully enabled for all $($totalSqlDatabaseWithTDEDisabled) SQL Server database(s)." -ForegroundColor $([Constants]::MessageType.Update)
+        }
+        else
+        {
+            Write-Host "Transparent Data Encryption (TDE) successfully enabled for $($($sqlServersRemediated | Measure-Object).Count) out of $($totalSqlDatabaseWithTDEDisabled) SQL Server database(s)." -ForegroundColor $([Constants]::MessageType.Warning)
+        }
+
+        
+        $colsProperty = @{Expression={$_.ServerName};Label="Server Name";Width=20;Alignment="left"},
+                        @{Expression={$_.DatabaseName};Label="Database Name";Width=20;Alignment="left"},
+                        @{Expression={$_.ResourceGroupName};Label="Resource Group Name";Width=20;Alignment="left"},
+                        @{Expression={$_.State};Label="State";Width=20;Alignment="left"}
+                              
+
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "Remediation Summary:`n" -ForegroundColor $([Constants]::MessageType.Info)
+        
+        if ($($sqlServersRemediated | Measure-Object).Count -gt 0)
+        {
+            Write-Host "Transparent Data Encryption (TDE) successfully enabled for the following SQL Server database(s):" -ForegroundColor $([Constants]::MessageType.Update)
+            $sqlServersRemediated | Format-Table -Property $colsProperty -Wrap
+
+            # Write this to a file.
+            $sqlServersRemediatedFile = "$($backupFolderPath)\RemediatedSQLDatabases.csv"
+            $sqlServersRemediated | Export-CSV -Path $sqlServersRemediatedFile -NoTypeInformation
+            Write-Host "This information has been saved to $($sqlServersRemediatedFile)"
+            Write-Host "Use this file for any roll back that may be required." -ForegroundColor $([Constants]::MessageType.Info)
+        }
+
+        if ($($sqlServersSkipped | Measure-Object).Count -gt 0)
+        {
+            Write-Host "Error occurred while enabling Transparent Data Encryption (TDE) for the following SQL Server database(s):" -ForegroundColor $([Constants]::MessageType.Error)
+            $sqlServersSkipped | Format-Table -Property $colsProperty -Wrap
+            
+            # Write this to a file.
+            $sqlServersSkippedFile = "$($backupFolderPath)\SkippedSqlServers.csv"
```

Comment:
Please have the same convention for the file name as in L392. And everywhere too!